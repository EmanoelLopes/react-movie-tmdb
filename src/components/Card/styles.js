import styled from 'styled-components';
import { colors, media, images } from 'utils/styled';

const { path, poster, placeholder } = images;

export const CardFeatured = styled.div`
  background-color: ${colors.white};
  background-image: url(
    ${props => (props.backdrop)
      ? `${path}${poster}${props.backdrop}`
      : `${placeholder.backdrop}`
    }
  );
  color: ${colors.white};

  ${media.desktop`
    background-image: none;
    color: ${colors.mainColor};
  `}

  > a {
    color: ${colors.white};
    display: flex;
    flex-wrap: wrap;
    height: 100%;
    text-decoration: none;

    ${media.desktop`
      color: ${colors.mainColor};
    `}
  }

  background-size: cover;
  background-position: top center;
  border: 1px solid #ccc;

  .card-poster figure,
  .card-poster img {
    height: 100%;
  }

  .card-poster img {
    display: none;
    width: 100%;

    ${media.desktop`
      display: block;
    `}
  }

  .card-info {
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    padding: 15px;

    ${media.desktop`
      justify-content: flex-start;
    `}
  }

  .card-info-wrapper {
    background-color: rgba(0,0,0,.7);
    padding: 10px;

    ${media.desktop`
      background-color: transparent;
      padding: 0;
    `}
  }

  .card-info h3 {
    font-size: 20px;
    margin-bottom: 15px;

    ${media.desktop`
      max-width: 200px;
      overflow: hidden;
      padding-bottom: 5px;
      white-space: nowrap;
      text-overflow: ellipsis;
    `}
  }
  .card-info small {
    display: block;
    margin-bottom: 15px;
  }

  .card-info p {
    margin-bottom: 15px;
  }

  .card-info a {
    border: 3px solid ${colors.white};
    color: ${colors.white};
    display: block;
    font-weight: 600;
    padding: 10px 15px;    
    width: 130px;
    text-align: center;
    text-decoration: none;

    ${media.desktop`
      color: ${colors.mainColor};
      border: 3px solid ${colors.mainColor};
    `}

    &:hover {
      text-decoration: uderline;
    }
  }
`;
