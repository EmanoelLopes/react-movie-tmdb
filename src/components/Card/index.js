import React from 'react';
import { string } from 'prop-types';
import intl from 'react-intl-universal';
import { Link } from 'react-router-dom';
import { CardFeatured } from './styles';
import { images } from 'utils/styled';
import { textTruncate } from 'utils/helpers';

const Card = props => {
  const {
    title,
    description,
    poster,
    id,
    backdrop,
  } = props;

  return (
    <CardFeatured backdrop={backdrop}>
      <Link to={id} title={`${title} - ${description}`}>
        <div className="card-poster">
          <figure>
            <img src={`${images.path}${images.poster}${poster}`} alt={title} lazyload="on"/>
          </figure>
        </div>
        <div className="card-info">
          <div className="card-info-wrapper">
            <h3>{title}</h3>
            <p>{
              (description) 
                ? <span>{textTruncate(description, 80)} <br /> [{intl.get('MORE_INFO')}]</span>
                : <span>({intl.get('NO_INFORMATION_PROVIDED')}) <br /> [{intl.get('MORE_INFO')}] </span>}
            </p>
          </div>
        </div>
      </Link>
    </CardFeatured>
  );
};

Card.propTypes = {
  poster: string,
  title: string,
  description: string,
  id: string,
  backdrop: string,
};

Card.defaultProps = {
  poster: '',
  title: '',
  description: '',
  id: '',
  backdrop: '',
};

export default Card;
