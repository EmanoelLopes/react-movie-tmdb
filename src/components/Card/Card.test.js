import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import Card from 'components/Card';
 
let wrapped;
let defaultProps = {
  title: 'Some Card',
  description: 'some description',
  poster: 'someposterpath.jpg',
  id: 'someid',
};

beforeEach(() => {
  wrapped = mount(
    <BrowserRouter>
      <Card {...defaultProps} />
    </BrowserRouter>
  );
});

afterEach(() => {
  wrapped.unmount();
});

describe('<Card />', () => {
  it('should match snapshot', () => {
    expect(wrapped).toMatchSnapshot();
  });

  it('should have a title h3', () => {
    expect(wrapped.find('.card-info h3')).toHaveLength(1);
  });

  it('should have a paragraph with description', () => {
    expect(wrapped.find('.card-info p')).toHaveLength(1);
  });

  it('should have a image tag with the poster path', () => {
    expect(wrapped.find('.card-poster figure img')).toHaveLength(1);
  });
});
