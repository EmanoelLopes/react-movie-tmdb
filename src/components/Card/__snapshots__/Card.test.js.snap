// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match snapshot 1`] = `
.c0 {
  background-color: #fff;
  background-image: url( https://placehold.jp/1280x720.png?text=NO%20IMAGE );
  color: #fff;
  background-size: cover;
  background-position: top center;
  border: 1px solid #ccc;
}

.c0 > a {
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  height: 100%;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 .card-poster figure,
.c0 .card-poster img {
  height: 100%;
}

.c0 .card-poster img {
  display: none;
  width: 100%;
}

.c0 .card-info {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 15px;
}

.c0 .card-info-wrapper {
  background-color: rgba(0,0,0,.7);
  padding: 10px;
}

.c0 .card-info h3 {
  font-size: 20px;
  margin-bottom: 15px;
}

.c0 .card-info small {
  display: block;
  margin-bottom: 15px;
}

.c0 .card-info p {
  margin-bottom: 15px;
}

.c0 .card-info a {
  border: 3px solid #fff;
  color: #fff;
  display: block;
  font-weight: 600;
  padding: 10px 15px;
  width: 130px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 .card-info a:hover {
  -webkit-text-decoration: uderline;
  text-decoration: uderline;
}

@media (min-width:62em) {
  .c0 {
    background-image: none;
    color: #081c24;
  }
}

@media (min-width:62em) {
  .c0 > a {
    color: #081c24;
  }
}

@media (min-width:62em) {
  .c0 .card-poster img {
    display: block;
  }
}

@media (min-width:62em) {
  .c0 .card-info {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media (min-width:62em) {
  .c0 .card-info-wrapper {
    background-color: transparent;
    padding: 0;
  }
}

@media (min-width:62em) {
  .c0 .card-info h3 {
    max-width: 200px;
    overflow: hidden;
    padding-bottom: 5px;
    white-space: nowrap;
    text-overflow: ellipsis;
  }
}

@media (min-width:62em) {
  .c0 .card-info a {
    color: #081c24;
    border: 3px solid #081c24;
  }
}

<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Card
      backdrop=""
      description="some description"
      id="someid"
      poster="someposterpath.jpg"
      release=""
      title="Some Card"
    >
      <styled.div
        backdrop=""
      >
        <StyledComponent
          backdrop=""
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": false,
                "lastClassName": "c0",
                "rules": Array [
                  "
  background-color: ",
                  "#fff",
                  ";
  background-image: url(
    ",
                  [Function],
                  "
  );
  color: ",
                  "#fff",
                  ";

  ",
                  "
    @media (min-width: ",
                  "62",
                  "em) {
      ",
                  "
    background-image: none;
    color: ",
                  "#081c24",
                  ";
  ",
                  "
    }
  ",
                  "

  > a {
    color: ",
                  "#fff",
                  ";
    display: flex;
    flex-wrap: wrap;
    height: 100%;
    text-decoration: none;

    ",
                  "
    @media (min-width: ",
                  "62",
                  "em) {
      ",
                  "
      color: ",
                  "#081c24",
                  ";
    ",
                  "
    }
  ",
                  "
  }

  background-size: cover;
  background-position: top center;
  border: 1px solid #ccc;

  .card-poster figure,
  .card-poster img {
    height: 100%;
  }

  .card-poster img {
    display: none;
    width: 100%;

    ",
                  "
    @media (min-width: ",
                  "62",
                  "em) {
      ",
                  "
      display: block;
    ",
                  "
    }
  ",
                  "
  }

  .card-info {
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    padding: 15px;

    ",
                  "
    @media (min-width: ",
                  "62",
                  "em) {
      ",
                  "
      justify-content: flex-start;
    ",
                  "
    }
  ",
                  "
  }

  .card-info-wrapper {
    background-color: rgba(0,0,0,.7);
    padding: 10px;

    ",
                  "
    @media (min-width: ",
                  "62",
                  "em) {
      ",
                  "
      background-color: transparent;
      padding: 0;
    ",
                  "
    }
  ",
                  "
  }

  .card-info h3 {
    font-size: 20px;
    margin-bottom: 15px;

    ",
                  "
    @media (min-width: ",
                  "62",
                  "em) {
      ",
                  "
      max-width: 200px;
      overflow: hidden;
      padding-bottom: 5px;
      white-space: nowrap;
      text-overflow: ellipsis;
    ",
                  "
    }
  ",
                  "
  }
  .card-info small {
    display: block;
    margin-bottom: 15px;
  }

  .card-info p {
    margin-bottom: 15px;
  }

  .card-info a {
    border: 3px solid ",
                  "#fff",
                  ";
    color: ",
                  "#fff",
                  ";
    display: block;
    font-weight: 600;
    padding: 10px 15px;    
    width: 130px;
    text-align: center;
    text-decoration: none;

    ",
                  "
    @media (min-width: ",
                  "62",
                  "em) {
      ",
                  "
      color: ",
                  "#081c24",
                  ";
      border: 3px solid ",
                  "#081c24",
                  ";
    ",
                  "
    }
  ",
                  "

    &:hover {
      text-decoration: uderline;
    }
  }
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c0"
          >
            <Link
              replace={false}
              to="someid"
            >
              <a
                href="/someid"
                onClick={[Function]}
              >
                <div
                  className="card-poster"
                >
                  <figure>
                    <img
                      alt="Some Card"
                      lazyload="on"
                      src="https://image.tmdb.org/t/p/w500someposterpath.jpg"
                    />
                  </figure>
                </div>
                <div
                  className="card-info"
                >
                  <div
                    className="card-info-wrapper"
                  >
                    <h3>
                      Some Card
                    </h3>
                    <small>
                      <FontAwesomeIcon
                        border={false}
                        className=""
                        fixedWidth={false}
                        flip={null}
                        icon={
                          Object {
                            "icon": Array [
                              448,
                              512,
                              Array [],
                              "f073",
                              "M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z",
                            ],
                            "iconName": "calendar-alt",
                            "prefix": "fas",
                          }
                        }
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        title=""
                        transform={null}
                      >
                        <svg
                          aria-hidden="true"
                          className="svg-inline--fa fa-calendar-alt fa-w-14 "
                          data-icon="calendar-alt"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 448 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
                            fill="currentColor"
                            style={Object {}}
                          />
                        </svg>
                      </FontAwesomeIcon>
                       
                    </small>
                    <p>
                      some description [more info]
                    </p>
                  </div>
                </div>
              </a>
            </Link>
          </div>
        </StyledComponent>
      </styled.div>
    </Card>
  </Router>
</BrowserRouter>
`;
